flink-java/src/main/java/org/apache/flink/api/java/operators/PartitionOperator.java

```diff
@@ -82,6 +88,7 @@ public PartitionOperator(DataSet<T> input, Keys<T> pKeys, Partitioner<?> customP
 		this.pKeys = pKeys;
 		this.partitionLocationName = partitionLocationName;
 		this.customPartitioner = customPartitioner;
+		this.distribution = distribution;
```

Comment:
We should check here that the key types of `distribution.getKeyTypes()` are equal to `pKeys.getKeyFieldTypes()`.